### PR TITLE
fix byte compilation warnings

### PR DESCRIPTION
**Pull request by markgdawson**
Minor fixes to remove byte compilation warnings on projectile.el

### CODE REVIEW

File: projectile.el [hunk 1]
```diff
@@ -1050,7 +1049,7 @@ discover projects there."
           (dolist (dir (directory-files directory t))
             (when (and (file-directory-p dir)
                        (not (member (file-name-nondirectory dir) '(".." "."))))
-              (projectile-discover-projects-in-directory dir (1- depth))))
+              (projectile-discover-projects-in-directory dir)))
```
Comment (bbatsov) [inline projectile.el:1052]: Why did you change this?

Comment (markgdawson) [reply projectile.el]: You're right. This shouldn't be here. Reverted.
